Switch to the new peer notify mechanism

Stebalien opened the pull request:
1. Only listen for peers added and identify events.
2. Remove the old "Limited" check. Peers only show up as "Connected" if they have non-limited connections.
3. Don't bother listening for new connections directly and/or connectivity changes. We'll get a new identify event per new connection regardless.

fixes #546

Stebalien committed:
Switch to the new peer notify mechanism

1. Only listen for peers added and identify events.
2. Remove the old "Limited" check. Peers only show up as "Connected" if
they have non-limited connections.
3. Don't bother listening for new connections directly and/or
connectivity changes. We'll get a new identify event per new connection
regardless.

fixes #546

Stebalien commented on peer_notify.go:
```diff
@@ -0,0 +1,91 @@
+)
+
+func (ps *PubSub) watchForNewPeers(ctx context.Context) {
+	// We don't bother subscribing to "connectivity" events because we always run identify after
+	// every new connection.
```
I _think_ this is right but I'm not 100% certain. I remember remember running into some issues with this with respect to bitswap...

cc @sukunrt do you think the code here is sufficient?

sukunrt replied:
Yeah this looks fine. We run identify after connecting on every connection.

vyzo reviewed (approved):
Looks reasonable.
In fact, I am quite happy we got rid of that ancient notify code!

Stebalien commented:
Apparently pubsub uses the "blank" host in tests, which doesn't support identify.

Stebalien commented:
Tests should be fixed by https://github.com/libp2p/go-libp2p-pubsub/pull/565.

Stebalien commented:
This conflicts with https://github.com/libp2p/go-libp2p-pubsub/pull/440. I'd prefer to just remove that feature because it doesn't work reliably with identify.

vyzo commented:
I agree, but we should check with users of the feature.
If there is none, no problem.

If it is in use, we might have to make some changes; maybe have a user function to test instead of directly comparing with the router protocols?

Stebalien commented:
Hm. You're right, we _have_ the function and can just use that.

Stebalien committed:
Use the "supports protocol" test function, if available

This is actually kind of backwards (it tells us if the peer would be
able to open a stream for us, not if we could open a steram to the peer)
but it's good enough.

Stebalien commented:
Ok, it should all work correctly-ish now. Or, well, correct enough.
